### ISSUE TITLE
I get an error when I execute: npm run lint:fix

### ISSUE THREAD

**Issue by Indomite** [typescript-eslint/typescript-eslint#5000]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

none

### Repro Code

```typescript
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.1.0

YOUR TYPESCRIPT VERSION: 4.6.4

Please only submit bug reports when using the officially supported version.
```

### ESLint Config

```javascript
module.exports = {
  rules: {
    // override/add rules settings here
  },
};
```

### Expected Result

I want to modify my format when I execute: npm run lint:fix

### Actual Result

reported an error

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^3.10.1` |
| `@typescript-eslint/parser`            | `^3.10.1` |
| `TypeScript`                           | `^4.0.0` |
| `ESLint`                               | `^6.8.0` || `^7.2.0` |
| `node`                                 | `16.14.1` |

**Comment by Josh-Cena** [comment]
> [x] I have updated to the latest version of the packages.

`^3.10.1` is apparently not the [latest version](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.24.0), though.

**Comment by Indomite** [comment]
> > * [x] 我已经更新到最新版本的软件包。
> 
> `^3.10.1`不过，显然不是[最新版本](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.24.0)。

I got it~
sorry，but I want to know which version of typescript-eslint should I install, when my typescript version is 4.6.4,thinks~

**Comment by Josh-Cena** [comment]
I've already given the link to 「最新版本」 😄 It's 5.24.0.

**Comment by Indomite** [comment]
> I've already given the link to 「最新版本」 😄 It's 5.24.0.

ok, i'll try it later.
Thanks again 😄
